### PR TITLE
add metrics

### PR DESCRIPTION
**Pull request by StevenTobin**
Add customMetric pkg to expose custom prometheus metrics for registry calls.

### CODE REVIEW

**Comment by laurafitzgerald** [comment]
Seeing as I was already chatting with you about this one @StevenTobin I'll wait for others to review before doing any review.
This is deployed here: https://master.lfitzger-3531.open.redhat.com/console/project/heimdall
Contact me for credentials.

**Review by maleck13 — COMMENTED** [review]
These changes look fine. However there is another set of registry calls here https://github.com/integr8ly/heimdall/blob/master/pkg/registry/client.go where it gets info about the image. We might want to record these also

File: pkg/customMetrics/customMetrics.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	RegistryCallsTotal = prometheus.NewCounter(
+		prometheus.CounterOpts{
+			Name: "registry_calls_total",
```
Comment (laurafitzgerald) [inline pkg/customMetrics/customMetrics.go]: @StevenTobin lets add a prefix to these. e.g. `heimdall_` so that they are more searchable in prometheus.

**Review by maleck13 — APPROVED** [review]
Looks good now. /lgtm will allow @laurafitzgerald to verify

**Comment by laurafitzgerald** [comment]
Will build and push the image to the pds for a final check and then we should be good to merge. ~There'll be a related pr to the integreatly installer for labelling the namespace which is required for prometheus to pick up the metrics.~ EDIT. No pr required. The label was missing in this case because it was installed seperate to the integreatly installation.

**Pull request merged**